### PR TITLE
Dont show spinner while offline and not focused

### PR DESCRIPTION
**Pull request by chrisnojima**
- [x] Move focus out of chat and into config. It's not just a chat thing
- [x] While not focused don't show the animation due to chrome cpu usage

@keybase/react-hackers

### CODE REVIEW

File: shared/reducers/chat.js [hunk 1]
```diff
@@ -164,9 +164,7 @@ function reducer (state: Constants.State = initialState, action: Constants.Actio
     }
     case 'chat:appendMessages': {
       const appendAction: Constants.AppendMessages = action
-      const appendMessages = appendAction.payload.messages
-      const isSelected = action.payload.isSelected
-      const conversationIDKey = appendAction.payload.conversationIDKey
+      const {messages: appendMessages, isSelected, conversationIDKey, isAppFocused} = appendAction.payload
```
Comment (gabriel) [inline shared/reducers/chat.js]: appFocused?

File: shared/actions/chat/creators.js [hunk 2]
```diff
@@ -185,8 +185,8 @@ function markSeenMessage (conversationIDKey: Constants.ConversationIDKey, messag
   return {payload: {conversationIDKey, messageID}, type: 'chat:markSeenMessage'}
 }
 
-function appendMessages (conversationIDKey: Constants.ConversationIDKey, isSelected: boolean, messages: Array<Constants.Message>): Constants.AppendMessages {
-  return {logTransformer: appendMessageActionTransformer, payload: {conversationIDKey, isSelected, messages}, type: 'chat:appendMessages'}
+function appendMessages (conversationIDKey: Constants.ConversationIDKey, isSelected: boolean, isAppFocused: boolean, messages: Array<Constants.Message>): Constants.AppendMessages {
```
Comment (gabriel) [inline shared/actions/chat/creators.js]: appFocused

File: shared/reducers/config.js [hunk 3]
```diff
@@ -190,6 +192,12 @@ export default function (state: ConfigState = initialState, action: Action): Con
       }
     }
 
+    case 'app:changedFocus':
+      return {
+        ...state,
+        appFocused: action.payload.focused,
```
Comment (gabriel) [inline shared/reducers/config.js]: maybe make the action.payload.appFocused to be consistent with same name everyhwere

**Comment by chrisnojima** [comment]
there is a chrome bug which still makes this run at 60 fps no matter what the gif framerate is

**Comment by gabriel** [comment]
oh hmm ok.. i actually modified the gif to be less frames and was able to have the electron renderer go to 0% but maybe there was a different scenario i missed

File: shared/reducers/chat.js (hunk 1 above)
Comment (chrisnojima) [inline shared/reducers/chat.js]: this is an unrelated action to the actual focusChanged one so i figured it should be more self similar (isSelected)

File: shared/reducers/config.js (hunk 3 above)
Comment (chrisnojima) [inline shared/reducers/config.js]: i can rename this one
